### PR TITLE
doc/releases: Add picolibc migration notes

### PR DESCRIPTION
**Pull request by keith-packard**
There aren't any 'required' or 'recommended' changes when switching from the minimal C library to picolibc, but there are a bunch of things which application developers might need to be aware of. Add a separate section to list those.

### CODE REVIEW

File: doc/releases/migration-guide-3.5.rst [hunk 1]
```diff
@@ -42,3 +42,42 @@ Recommended Changes
   :kconfig:option:`CONFIG_GIC_V3` directly in Kconfig has been deprecated.
   The GIC version should now be specified by adding the appropriate compatible, for
   example :dtcompatible:`arm,gic-v2`, to the GIC node in the device tree.
+
+Picolibc-related Changes
```
Comment (fabiobaltieri) [inline doc/releases/migration-guide-3.5.rst:46]: Should there be something about `CBPRINTF_FP_SUPPORT`? I had to manually set it off in one of my downstream applications to match the previous firmware size and I think it's been discussed in TSC for some other application as well.

Comment (keith-packard) [reply doc/releases/migration-guide-3.5.rst:46]: I'm not familiar with the details of that -- can you provide some suggested wording?

Comment (fabiobaltieri) [reply doc/releases/migration-guide-3.5.rst:46]: It was discussed here: https://github.com/zephyrproject-rtos/zephyr/pull/57340#issuecomment-1551945883, maybe something on the line of picolibc providing a complete cbprintf floating point support by default at the expense of flash size, but the feature can be disabled by tweaking that flag.

Comment (keith-packard) [reply doc/releases/migration-guide-3.5.rst:46]: Thanks for the reminder. I've added an item which describes this issue.

Comment (fabiobaltieri) [reply doc/releases/migration-guide-3.5.rst:46]: Good stuff, thanks!

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@kartben ping

File: doc/releases/migration-guide-3.5.rst [hunk 2]
```diff
@@ -42,3 +42,51 @@ Recommended Changes
...
+
+* Picolibc can either be built as part of the OS build or pulled from the
+  toolchain. When building as part of the OS, the build will increase by
+  approximately 1000 files.
+
+* When using the standard C++ library with Picolibc, both of those must come
+  from the toolchain as the standard C++ library depends upon the C library
+  ABI.
+
+* Picolibc removes the `-ffreestanding` compiler option. This allows
```
Comment (kartben) [inline doc/releases/migration-guide-3.5.rst:80]:
```suggestion
* Picolibc removes the ``-ffreestanding`` compiler option. This allows
```

File: doc/releases/migration-guide-3.5.rst [hunk 3]
```diff
@@ -42,3 +42,51 @@ Recommended Changes
...
+  significant compiler optimization improvements, but also means that the
+  compiler will now warn about declarations of `main` which don't conform to
+  the Zephyr required type -- `int main(void)`.
```
Comment (kartben) [inline doc/releases/migration-guide-3.5.rst:83]:
```suggestion
  the Zephyr required type -- ``int main(void)``.
```

**Review by kartben — CHANGES_REQUESTED** [review]
Only two cosmetic/styling issues.
The text is great, and I like the idea of having a dedicated section for it, but we'll have to see how the overall structure of the migration guide might need to evolve as it becomes more crowded closer to the release date. // @carlescufi

File: doc/releases/migration-guide-3.5.rst (hunk 3 above)
Comment (keith-packard) [reply doc/releases/migration-guide-3.5.rst]: thanks; there are too many text formatting conventions :-)

**Review by kartben — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**